docs/components.md

```diff
@@ -332,6 +333,50 @@ $ cat kubeprod-manifest.jsonnet
 }
 ```
 
+### Grafana
+
+[Grafana](http://docs.grafana.org/) is an open source metric analytics & visualization suite. It is most commonly used for visualizing time series data for infrastructure and application analytics but many use it in other domains including industrial sensors, home automation, weather, and process control.
+
+#### Implementation
+
+It runs on top of Kubernetes as a [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) with just 1 non-root pod named `grafana-0` under the `kubeprod` namespace and also provides a Kubernetes Ingress resource named `grafana` which allows end-user access to Grafana from the Internet. It is implemented inside the [Grafana](../manifests/components/grafana.jsonnet) manifest.
+
+#### Authentication
+
+Grafana delegates authentication to [OAuth2 Proxy](#oauth2-proxy). Once authenticated, the user is allowed access to Grafana as an administrator.
+
+#### Configuration
+
+Grafana ships with a default configuration that configures Prometheus as a datasource for Grafana. This configuration is implemented by the `datasources` ConfigMap resource which defines a single datasource named `BKPR Prometheus` pointing to BKPR's Prometheus instance.
+
+##### Networking
+
+Grafana exposes port `3000/tcp` internally to the Kubernetes cluster, but allows external access via HTTP/S by means of the deployed nginx-ingress-controller. Grafana can connect to Prometheus via a datasource that ships with the default configuration.
+
+##### Storage
+
+To assure the resilience of Grafana, each pod relies on a Kubernetes PersistentVolume named `datadir-grafana-%i` where `%i` is an index that matches the pod index. By default, each PersistentVolume is allocated a default storage of 1GiB. In the Overrides section below there are instructions for reconfiguring this.
```

Comment:
What do you mean by `resilience of Grafana`? The persistent volume persists the state of the grafana pod's beween upgrades / restarts right? When you say resilience it appears you're talking about HA.

Reply:
I have rephrased it.